Remove all references to 'grained' access control, and reword a few t… (#7946)
auth0/docs: articles/architecture-scenarios/_includes/_authorization/_introduction.md

> It’s important to start by distinguishing between Authentication, Authorization, and Access Control. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and some or all of Authorization and Access Control. Additionally, ab Application or API itself may need to share Access Control responsibility, especially where contextual access is required: 

Author: ```suggestion
It’s important to start by distinguishing between Authentication, Authorization, and Access Control. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and some or all of Authorization and Access Control. Additionally, an Application or API itself may need to share Access Control responsibility, especially where contextual access is required: 
```

* **Authentication**: the process of determining if a principal - a user or application - is who/what they say they are.
* **Authorization**: the proceess of determining what is allowed, based on the principal, the consent they provide, what permissions they have been given, and/or the set of contextually specific access criteria.
* **Consent**: what can be done on the behalf of a user principal (subject) or, in the case of an application principal authenticating via [Client Credentials](/flows/concepts/client-credentials) grant, what an administrative authority has authorized an application to do.

> * **Access Control**: the access that is permitted. In general we typically group different types of access control into three distinct categories, so that it's easier to understand (a) which actor is responsible for storing the information, (b) which actor is responsible for making decisions, and (c) which is responsible for enforcing the restrictions. 

Author: ```suggestion
* **Access Control**: Access control is really the umbrella concept that puts Authentication, Authorization, and Enforcement into one bucket.  In general we typically group different types of access control into three distinct categories, so that it's easier to understand (a) which actor is responsible for storing the information, (b) which actor is responsible for making decisions, and (c) which is responsible for enforcing the restrictions. 
```

> * **Access Control**: the access that is permitted. In general we typically group different types of access control into three distinct categories so that it's easier to understand a) which actor is responsible for storing the information, b) which actor is responsible for making decisions, and c) which is responsible for enforcing the restrictions. 

Author: ```suggestion
* **Access Control**: This is an umbrella concept that ties together Authentication, Authorization, and Enforcement of policies.  In general we typically group different types of access control into three distinct categories so that it's easier to understand a) which actor is responsible for storing the information, b) which actor is responsible for making decisions, and c) which is responsible for enforcing the restrictions. 
```

> Let's start by taking a step back and talking about Access Control.  There isn't one clear cut definition of Access Control in the industry, but if you spend some time searching and reading you'll see that most authoritative sources agree that it is the umbrella concept that puts all of Authentication, Authorization, Consent, and Policy Enforcement together to ensure that only the right people and services have access to your applications and APIs.  Next, let's dive more closely into the distinctions between Authentication, Authorization, Consent, and Policy Enforcement. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and Consent, and some or all of Authorization and Policy Enforcement. Additionally, an Application or API itself almost always is the primary enforcer of policies, especially where contextual access is required: 

Reviewer: Does one "dive more closely"? I'd say perhaps one would "dive deeper" or "look more closely"?

Author: ```suggestion
Let's start by taking a step back and talking about Access Control.  There isn't one clear cut definition of Access Control in the industry, but if you spend some time searching and reading you'll see that most authoritative sources agree that it is the umbrella concept that puts all of Authentication, Authorization, Consent, and Policy Enforcement together to ensure that only the right people and services have access to your applications and APIs.  Next, let's look more closely into the distinctions between Authentication, Authorization, Consent, and Policy Enforcement. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and Consent, and some or all of Authorization and Policy Enforcement. Additionally, an Application or API itself almost always is the primary enforcer of policies, especially where contextual access is required: 
```

> Let's start about by taking a step back and talking about Access Control.  There isn't one clear cut definition of Access Control in the industry, but if you spend some time searching and reading you'll see that most authoritative sources agree that it is the umbrella concept that puts all of Authentication, Authorization, Consent, and Policy Enforcment together to ensure that only the right people and services have access to your applications and APIs.  Next, let's dive more closely into the distinctions between Authentication, Authorization, Consent, and Policy Enforcement. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and Consent, and some or all of Authorization and Policy Enforcement. Additionally, an Application or API itself almost always is the primary enforcer of policies, especially where contextual access is required: 

Reviewer 2: misspelled enforcement in the first instance here

> Let's start about by taking a step back and talking about Access Control.  There isn't one clear cut definition of Access Control in the industry, but if you spend some time searching and reading you'll see that most authoritative sources agree that it is the umbrella concept that puts all of Authentication, Authorization, Consent, and Policy Enforcment together to ensure that only the right people and services have access to your applications and APIs.  Next, let's dive more closely into the distinctions between Authentication, Authorization, Consent, and Policy Enforcement. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and Consent, and some or all of Authorization and Policy Enforcement. Additionally, an Application or API itself almost always is the primary enforcer of policies, especially where contextual access is required: 

Author: ```suggestion
Let's start by taking a step back and talking about Access Control.  There isn't one clear cut definition of Access Control in the industry, but if you spend some time searching and reading you'll see that most authoritative sources agree that it is the umbrella concept that puts all of Authentication, Authorization, Consent, and Policy Enforcement together to ensure that only the right people and services have access to your applications and APIs.  Next, let's dive more closely into the distinctions between Authentication, Authorization, Consent, and Policy Enforcement. Your Auth0 tenant (your Authorization Server) is typically responsible for Authentication and Consent, and some or all of Authorization and Policy Enforcement. Additionally, an Application or API itself almost always is the primary enforcer of policies, especially where contextual access is required: 
```

    
* The third category is where access is either granted or denied depending on what the principal (subject) can operate on within the context of an application or API. Both the data required to enforce this, and the enforcement process is typically defined in the context of the application or API. In this scenario, the data communicated as one or more custom claims in an [`id`](/tokens/id-token) or [`access`](/tokens/overview-access-tokens) token may be consumed with or without data from an external source that is not Auth0.

In addition, Role-based Access Control (RBAC) and Attribute-based Access Control (ABAC) mechanisms can be applied in any of the Access Control categories described above. Whatever your use case then, there are a number of things you will want to consider when looking at the functionality and workflow you require:

* Are there scenarios where access to an entire application or API should be rejected?
* Will you be providing APIs that can be accessed by third-party applications?
* Will your APIs also be accessed by your own (first-party) applications?
* Will your application be calling a third-party API?

> * Should your applications and/or APIs be enforcing access control based on user information?

Reviewer 2: user "claims" - to align with SAML and OIDC terminology?

Author: ```suggestion
* Should your applications and/or APIs be enforcing access control based on user claims?
```

::: panel Best Practice
In the main, we have found that [OIDC](/protocols/oidc) is the most commonly used industry-standard protocol used by Auth0 customers when it comes to authentication in their applications. We have also found that, even though [OAuth2](protocols/oauth2) was created as a delegation protocol, it is commonly used within first party applications when there is an API that does not have a shared session with the application.
:::

> Auth0 also can provide the information needed so that an application or API can enforce restrictions. For [application level integration](#application-integration), Auth0 allows you to add [custom claims](#id-token-claims) to an ID Token, which your application can then verify and subsequently use to enforce access control. In this case you will need to decide what information you require for your application to make access control decisions.

Author: ```suggestion
Auth0 also can provide the information needed so that an application can enforce restrictions. For [application level integration](#application-integration), Auth0 allows you to add [custom claims](#id-token-claims) to an ID Token, which your application can then verify and subsequently use to perform policy enforcement. In this case you will need to decide what information you require for your application to make enforcement decisions.  If you need to make decisions at an API instead of in your application, you will likely need to use an Access Token instead of an ID token.  Continue reading for more information.
```

> 

Author: ```suggestion
```

> ::: warning

Author: ```suggestion
```

> ID Tokens should only be used within the application itself, it should not be used as a Bearer token to call an API.  If you need a token to call an API you should request an Access token instead.  See [API level integration](#api-integration) for more information.

Author: ```suggestion
```

> :::

Author: ```suggestion
```

::: warning

> When deciding what data to include in your ID token and/or access token, consider token size, especially if you are passing the token in the URL. Even if you are not passing tokens in the URL, you will also need to consider the potential of exposing sensitive PII (Personally Identifiable Information). Token information is not encrypted, so although it isn't generally a security issue for an ID token to be leaked, it can become a privacy issue depending on the data that is included in the token.

Reviewer: Back-to-back warning panels feels a little "clunky" from a UX perspective.

Author: removing first warning in lieu of inline comments.
